python/hail/table.py

```diff
@@ -372,11 +372,25 @@ def count(self):
     def _force_count(self):
         return self._jt.forceCount()
 
-    @typecheck_method(caller=str, s=expr_struct())
-    def _select(self, caller, s):
-        base, cleanup = self._process_joins(s)
-        analyze(caller, s, self._row_indices)
-        return cleanup(Table(base._jt.select(s._ast.to_hql())))
+    @typecheck_method(caller=str, key_struct=nullable(expr_struct()), value_struct=nullable(expr_struct()))
+    def _select(self, caller, key_struct=None, value_struct=None):
+        if key_struct is None:
+            assert(value_struct is not None)
```

Comment:
unnecessary parens

Reply:
how does that work? just `assert value_struct is not None`?

Reply:
yep